### DISCUSSION TITLE
result mp4 files after searching cannot be watched

### DISCUSSION THREAD

**Discussion by phuvinhnguyen** [vitrivr/cineast#344] [Q&A]
Dear devs,

I am having some problems when the result mp4 files last for 0 seconds and cannot be watched. I think this is because my extraction_config.json file only exports thumbnails, which then be used in mp4 result files.

I hope you can help me change those mp4 files to pictures or make them back to normal.

Thank you.

This is my extraction_config.json file
[extraction_config.json.txt](https://github.com/vitrivr/cineast/files/9776514/extraction_config.json.txt)

**Comment by silvanheller** [comment]
Hey, I'm not sure I understand your question. Thumbnails are objects are separate things in vitrivr, so the UI expects also access to the original objects. Cineast does not transform or copy the original objects, so you need to store these manually somewhere accessible.

**Reply by sauterl** [reply]
One way to do what @silvanheller mentioned is by specifying in the configuration where the original media (I guess in your case videos) are and let them be served by cineast.

Alternatively you can use a web server (e.g. nginx) to serve the files

Here is some relevant info in the wiki:

https://github.com/vitrivr/cineast/wiki/Retrieval-Setup-Guide#full-stack-vitrivr

**Reply by phuvinhnguyen** [reply]
This is the mp4 result after searching using vitrivr-ng, v_0000001_1 is a jpg file in thumbnails folder
![image](https://user-images.githubusercontent.com/81418873/195614561-edd9e042-055b-477d-8b5b-bbc234bed2c2.png)

I dont know how to fix this problem.

**Reply by silvanheller** [reply]
What is the url of the video file? That is, if you right-click on the video file and open it in a new tab, which url does vitrivr try to access?

**Reply by phuvinhnguyen** [reply]
it is "http://localhost:4200/", I am not sure if that is what you ask for.

This is the detail of that file
![image](https://user-images.githubusercontent.com/81418873/195623832-25df3f14-9aa7-4de3-aaf6-6e90691124b5.png)

**Reply by phuvinhnguyen** [reply]
also, when I click on the path, the file can not be downloaded(no file).

**Reply by silvanheller** [reply]
if you right-click on the media player, every modern browser should give you a context menu on which you can either open the video in a new tab (which shows you the url) or copy the url, see e.g. 
![image](https://user-images.githubusercontent.com/14314470/195625796-df7fb947-2410-46be-8b87-7bebcb272a8a.png)

**Reply by phuvinhnguyen** [reply]
it is http://localhost:4200/objects/v_0000001

**Reply by silvanheller** [reply]
Are you using a customized config file for vitrivr-ng? Additionally, can you try accessing localhost:4567/objects/v_0000001 via a browser and let us know if that works for you.

**Reply by phuvinhnguyen** [reply]
it works, I mean I can find the video at localhost:4567/objects/v_0000001, however, I don't know to redirect vitrivr-ng to this address

**Reply by phuvinhnguyen** [reply]
yeah, I do change some on the vitrivr-ng config file. However, I am not sure why it does not work. At the beginning, I don't have that file in src or dist folder but this problem still happen
[config.json.txt](https://github.com/vitrivr/cineast/files/9777915/config.json.txt)

**Reply by silvanheller** [reply]
Paging @sauterl, I think this is a bug in the config-parsing

**Reply by sauterl** [reply]
While I admit, the config parsing for resources can be improved, I suggest that you, @phuvinhnguyen , try the following config (adjusted from yours):

```
{
  "title": "Custom vitrivr",
  "api": {
  },
  "resources": {
    "host_thumbnails": "http://localhost:4567/exporter/thumbnails/:s",
    "host_objects": "http://localhost:4567/objects/:o",
    "port": "4567"
  },
  "competition": {
    "host": "",
    "tls": false,
    "log": false,
    "lsc": false,
    "vbs": false,
    "textualInput": false
  },
  "mlt": {
    "VIDEO": [
      "visualtextcoembedding"
    ],
    "IMAGE": [
      "visualtextcoembedding"
    ]
  },
  "query": {
    "history": 0,
    "options": {
      "image": true,
      "audio": false,
      "model3d": false,
      "text": true,
      "tag": false,
      "map": false,
      "semantic": true,
      "boolean": true,
      "skeleton": false
    },
    "metadata": {
      "object": [
        ["technical", "duration"]
      ],
      "segment": [
        ["*", "*"]
      ]
    },
    "text": {
      "categories": [
        [
          "visualtextcoembedding",
          "Text Co-Embedding"
        ]
      ]
    },
    "boolean": [
      {
        "display": "Segment Id",
        "input": "TEXT",
        "table": "cineast_segment",
        "col": "segmentid",
        "operators": [
          "="
        ]
      },
      {
        "display": "Object Id",
        "input": "TEXT",
        "table": "cineast_segment",
        "col": "objectid",
        "operators": [
          "="
        ]
      }
    ]
  },
  "refinement": {
    "filters": [
      [
        "dominantcolor.color",
        "CHECKBOX"
      ],
      [
        "technical.duration",
        "SLIDER"
      ]
    ]
  }
}
```

**Reply by phuvinhnguyen** [reply]
it still connects to port 4200 and the video cannot be watched
![image](https://user-images.githubusercontent.com/81418873/195837457-0a154509-0d8f-48a2-80fa-587f5b74dd9b.png)

**Reply by sauterl** [reply]
Sadly, then there is more broken. I'll have to look into that...
My apologies!

**Reply by phuvinhnguyen** [reply]
Then how can I change the default port to data in vitrivr-ng code ?

**Comment by phuvinhnguyen** [comment]
I have just found out that instead of using http://localhost:4200/ as it appears when I run 'ng serve' on vitrivr-ng, using http://localhost:4567/ can work as normal.
